recipe/meta.yaml

```diff
@@ -85,71 +83,168 @@ requirements:
     - qt ==5.15.2           # [arm64 or aarch64]
     - zlib
 
-test:
-  requires:
-    - {{ compiler('cxx') }}
-    - cmake
-    - python
-    - numpy
-  files:
-    - test-cmake/CMakeLists.txt
-    - test-cmake/DisplayImage.cpp
-    - test.cpp
-  commands:
-    - pushd test-cmake
-    - cmake . -DOpenCV_STATIC=OFF -DOpenCV_SHARED=ON -G"Visual Studio 14 2015"        # [win32]
-    - cmake . -DOpenCV_STATIC=OFF -DOpenCV_SHARED=ON -G"Visual Studio 14 2015 Win64"  # [win64]
-    - cmake . -DOpenCV_STATIC=OFF -DOpenCV_SHARED=ON                                  # [not win]
-    - cmake --build . --config Release
-    - popd
-    # Verify dynamic libraries.
-    # "bioinspired", Not working in 3.1.0
-    {% set opencv_libs = [
-         "aruco",
-         "bgsegm",
-         "calib3d",
-         "ccalib",
-         "core",
-         "datasets",
-         "dpm",
-         "face",
-         "features2d",
-         "flann",
-         "fuzzy",
-         "highgui",
-         "imgcodecs",
-         "imgproc",
-         "line_descriptor",
-         "ml",
-         "objdetect",
-         "optflow",
-         "phase_unwrapping",
-         "photo",
-         "plot",
-         "reg",
-         "rgbd",
-         "saliency",
-         "shape",
-         "stereo",
-         "stitching",
-         "structured_light",
-         "superres",
-         "surface_matching",
-         "text",
-         "tracking",
-         "video",
-         "videoio",
-         "videostab",
-         "xfeatures2d",
-         "ximgproc",
-         "xobjdetect",
-         "xphoto",
-    ] %}
-    {% for each_opencv_lib in opencv_libs %}
-    - test -f $PREFIX/lib/libopencv_{{ each_opencv_lib }}.dylib                        # [osx]
-    - test -f $PREFIX/lib/libopencv_{{ each_opencv_lib }}.so                           # [linux]
-    - if not exist %PREFIX%\\Library\\bin\\opencv_{{ each_opencv_lib }}342.dll exit 1  # [win]
-    {% endfor %}
+
+outputs:
+  - name: opencv
```

Comment:
It seems odd to me that the libopencv and py-opencv packages would depend on the opencv package and not the other way around. I'm sure there is a good reason, could you add some comments to the recipe explaining the rationale behind these spilts and the dependency handling for other maintainers? Thanks!

Reply:
This is based on a decision to merge libopencv (compiled libraries) and py-opencv (python stack which has dependencies on the compiled libraries). The separation was judged to add excessive maintainence overhead without sufficient benefit. The two additional empty packages are added to keep backwards compatibility with any downstream recipes that made use of either package.

I will confirm that this is indeed implemented as we wish it to be and add an explanatory note for maintainers.